File: js-jasmine/src/agedBrie.js
const Item = require('../src/item.js')
class AgedBrie {
    constructor(items) {
        this.items = items
    }

    updateQuality() {
        if (this.items.quality < 50) {
            this.items.quality += 1
        }
        if (this.items.sellIn !== 0) {
            this.items.sellIn -= 1
        }
        return this.items
    }
}

module.exports = AgedBrie

File: Java/src/main/java/com/gildedrose/items/AgedBrie.java
package com.gildedrose;

public class AgedBrie {
    public Item update(Item cheese){
        if(cheese.sellIn > 0){
            cheese.quality = Math.min(50, cheese.quality += 1);
        }else{
            cheese.quality = Math.min(50, cheese.quality += 2);
        }
        cheese.sellIn --;
        return cheese;
    }
}
